File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.cpp
class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& nums, int k) {

        unordered_map<int, int> mp;
        for(int i = 0; i < nums.size(); i++){
            int key = nums[i];

            if(mp.count(key) > 0){
                int j = mp[key];
                if(abs(i - j) <= k)
                    return true;
            }

            mp[key] = i;
        }
        return false; 
    }
};

File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.java
class Solution {

    public boolean containsNearbyDuplicate(int[] nums, int k) {
        int n = nums.length;
        //Map<Integer, Integer> mp = new HashMap<>();
        Set<Integer> set = new HashSet<>();

        for(int i = 0; i < n; i++){
            // if(mp.containsKey(nums[i]) && Math.abs(i - mp.get(nums[i])) <= k)
            //     return true;          
            // mp.put(nums[i], i);
            
            if(set.contains(nums[i]))
                return true;
            
            set.add(nums[i]);
            if(set.size() > k)
                set.remove(nums[i-k]);
        }

        return false;
    }
}